### DISCUSSION TITLE
How do I find the model attributes for the elements in the dataset?

### DISCUSSION THREAD

**Discussion by vsa1920** [deepmodeling/deepmd-kit#2575] [Q&A]
I am working on training on Cu-dataset as provided [here](https://www.aissquare.com/datasets/detail?pageType=datasets&name=Cu-dpgen). 

They also have a pretrained model available [here](https://www.aissquare.com/models/detail?pageType=models&name=Cu-dpgen-model). 

I am sorry, I am a newbie at both running LAMMPS and using DeePMD. Whenever I try to run a LAMMPS simulation using these models, I get this error:

```
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
  >>> Info of model(s):
  using   1 model(s): Cu-dpgen-updated.pb 
  rcut in model:      8
  ntypes in model:    1
ERROR: pair_coeff: element Cu not found in the model (/project/source/lmp/pair_deepmd.cpp:1100)
Last command: pair_coeff     * * Cu
```

I have been trying to debug this issue for a week now. Any help would be greatly appreciated. Thanks!

**Comment by njzjz** [comment]
The elements must be defined during the training time to map the elements, otherwise, only the default order can be used. Please read the documentation for details.

Probably, this model wasn't given the type map. CC @AnguseZhang

**Reply by vsa1920** [reply]
Hi! I have been stuck on this for quite long and started working on another project. Cracking my head on the documentation hasn't helped very much. So for starters:

1) Where exactly do we specify type_map? I specified one in the model training input script and that did not really help.
2) Is there some kind of workshop or maybe video resources to the entire process? That would be really helpful!

**Reply by njzjz** [reply]
It is specified in the input script, as you said. Did you use your model? In your original question, it seems that you used others' models.

Princeton University has hosted several workshops. See https://github.com/CSIprinceton

**Reply by vsa1920** [reply]
Earlier, I did. Then I retrained a model and tried to use it for a LAMMPS simulation, but it didn't work out. I am still trying out some stuff.

Thank you for the Princeton workshop reference, I will check it out.

**Reply by njzjz** [reply]
You can also remove atomic types from `pair_coeff`, i.e. `pair_coeff * *`, if the type information is not given by the model. The types will be assumed to be the same as training data.

**Reply by njzjz** [reply]
Could you provide `input.json` and the version you use to help locate the problem?

**Reply by vsa1920** [reply]
Thank you so much for your help!
Here's a link to input.json: https://drive.google.com/open?id=1cIdK1-dFkwQ7n8FCqWc_5Q-jqbY1wk-4&usp=drive_copy

**Reply by njzjz** [reply]
I don't have access to this file.

**Comment by vsa1920** [comment]
> You can also remove atomic types from `pair_coeff`, i.e. `pair_coeff * *`, if the type information is not given by the model. The types will be assumed to be the same as training data.

I got this error when I tried:

Generated 0 of 0 mixed pair_coeff terms from geometric mixing rule
ERROR: Not all per-type masses are set. Type 1 is missing. (src/src/atom.cpp:726)
Last command: run 5000

**Reply by njzjz** [reply]
It's just a LAMMPS error. It seems that you don't set the mass for atomic types.

**Comment by vsa1920** [comment]
> It's just a LAMMPS error. It seems that you don't set the mass for atomic types.

Hi! Thanks again for your help. It's invaluable for someone who is new to both LAMMPS and DeePMD. I am trying to run this on a local GPU cluster. Earlier, I was able to run this on Colab GPU but that was still incredibly slow ~ 500 LAMMPS steps in a small system size in 8 minutes. I am getting this error:

2023-07-31 15:59:35.884004: E tensorflow/core/common_runtime/session.cc:91] Failed to create session: INTERNAL: cudaGetDevice() failed. Status: cudaGetErrorString symbol not found.
INTERNAL: cudaGetDevice() failed. Status: cudaGetErrorString symbol not found.
ERROR on proc 0: DeePMD-kit C API Error: DeePMD-kit Error: TensorFlow Error: INTERNAL: cudaGetDevice() failed. Status: cudaGetErrorString symbol not found. (/project/source/lmp/pair_deepmd.cpp:838)
Last command: pair_style     deepmd Cu-dpgen-model-updated.pb

Any thoughts on this?

**Reply by njzjz** [reply]
It should be CUDA version mismatch. What's your CUDA version and how did you install deepmd-kit?

**Reply by vsa1920** [reply]
I used easy install using pip to install deepmd; using - pip install deepmd-kit[gpu,cu11,lmp]
The cuda version on the machine is 12.0, same as on colab. I checked through nvidia-smi.

**Reply by njzjz** [reply]
The laster stable TF version only supports CUDA 11.x. You can check `LD_LIBRARY_PATH` and see whether CUDA 12.0 is loaded.

**Comment by vsa1920** [comment]
I was checking the CUDA version from LD_LIBRARY_PATH and its 11.0. I am wondering if I did it correctly:

![Screenshot 2023-07-31 at 6 12 18 PM](https://github.com/deepmodeling/deepmd-kit/assets/77106994/78926944-6119-4acf-a021-6cd532ddc009)
